### PR TITLE
[4.0] Blocked user logout action should not be Logged when user is not logged in Frontend

### PR DESCRIPTION
**Pull request by sudhir-j-sapkal**
### Summary of Changes
If user is blocked , Logout activity should not be logged.

### Testing Instructions
Login to Joomla Backend with Superuser access

Go to Users => Manage
Disable any user
Go to Users => Users Action Log
See the latest activity. "User vjtest logged out from admin"

### Expected result

The logout entry for blocked user should not be logged when user is not logged in Front end, It should only log user is updated by Super user.

### Actual result

There were two activities 
1. For update of user by super user.
2. User is logged out from system.

screenshot-vijayktest joomla com-2019 10 19-17_27_56
screenshot-vijayktest joomla com-2019 10 19-17_26_53 (1)

### Documentation Changes Required
No

### CODE REVIEW

File: plugins/actionlog/joomla/joomla.php [hunk 1]
```diff
@@ -847,6 +847,11 @@ public function onUserLogout($user, $options = array())
 		}
 
 		$loggedOutUser      = User::getInstance($user['id']);
```
Comment (manojLondhe) [inline plugins/actionlog/joomla/joomla.php:849]: change as

```
		$loggedOutUser = User::getInstance($user['id']);
		
		if ($loggedOutUser->block)
		{
			return;
		}

		$messageLanguageKey = 'PLG_ACTIONLOG_JOOMLA_USER_LOGGED_OUT';
```

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: plugins/actionlog/joomla/joomla.php (hunk 1 above)
Comment (sudhir-j-sapkal) [reply plugins/actionlog/joomla/joomla.php:849]: @manojLondhe DONE

**Comment by patjun** [comment]
I have tested this item :white_check_mark: successfully on e6655ea2cfe5376e72fc45c27313d3c565e010b2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by brianteeman** [comment]
Why should this not be logged. It seems sensible to me that this is logged

**Comment by karo3** [comment]
I have tested this item :white_check_mark: successfully on e6655ea2cfe5376e72fc45c27313d3c565e010b2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by karo3** [comment]
It's only happens when you block him from the table but not when you do it in the edit view.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by uglyeoin** [comment]
@karo3 this should not be a successful test then

**Comment by vijaykhollam** [comment]
I have tested this item :white_check_mark: successfully on e6655ea2cfe5376e72fc45c27313d3c565e010b2<br /><br /><hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by sudhir-j-sapkal** [comment]
@brianteeman - It should log the activity of User Logged out only when user is log-ed in front end and Admin is blocking him from back-end.

**Comment by tushar33** [comment]
I have tested this item :white_check_mark: successfully on e6655ea2cfe5376e72fc45c27313d3c565e010b2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26688">issues.joomla.org/tracker/joomla-cms/26688</a>.</sub>

**Comment by brianteeman** [comment]
@sudhir-j-sapkal sorry but I stilll dont understand why this is needed

**Comment by mbabker** [comment]
It seems like there should be a more appropriate "Blocked user X tried to log in" log message, not no log message at all.  I'm with Brian here, I don't get what cognitive benefit this change is adding.

**Comment by Quy** [comment]
Closing for reason stated.

**Pull request closed without merging**